### ISSUE TITLE
Screen Reader: Profile Subsections: No feedback announced after requesting a new OTP code

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77573]
## Action Performed:
Precondition: User is logged in

1. Using VoiceOver on iOS or TalkBack on Android, open the app
2. Swipe right to focus the Account tab and double-tap to activate it
3. Swipe right to focus the Profile button and double-tap to activate it
4. Swipe right to focus the Contact methods menu item and double-tap to activate it
5. Swipe right to focus an email address that requires verification and double-tap to activate it
6. Swipe right to focus the "Didn't receive a code?" button and double-tap to activate it
7. Note if the new OTP code request is announced

## Expected Result:
After requesting a new OTP code screen readers should announce a confirmation message, e.g., "A new code has been sent to your email".

Users should know that the action succeeded and can proceed to enter the new OTP.

## Actual Result:
No feedback is announced after requesting a new OTP code.

Users relying on screen readers cannot confirm that the code was sent.

## Area issue was found in:
Profile

## Failed WCAG checkpoints
WCAG 4.1.3 Status Messages
https://www.w3.org/WAI/WCAG22/Understanding/status-messages

## User impact:
When appropriate roles or properties are assigned to status messages, the new content is spoken by screen readers in such a way as to assist blind and low vision users. Assigning proper roles or properties to status messages provides possible future uses and personalization opportunities, such as the potential to be exploited by assistive technologies created for users with some cognitive disabilities.

## Suggested resolution:
Use accessibilityLiveRegion on Android / AccessibilityInfo.announceForAccessibility on iOS to ensure screen reader users receive feedback announcement after requesting a new OTP code, e.g., "A new code has been sent to your email".
https://appt.org/en/docs/react-native/samples/accessibility-live-region
https://reactnative.dev/docs/accessibilityinfo#announceforaccessibility

## Environment:
Apple iPhone 14 Pro - iOS 18.7.1, Wi-Fi - Native (No mobile browser)
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

<img width="1179" height="2438" alt="Image" src="https://github.com/user-attachments/assets/22b3e726-984e-4595-973f-08e1af550a56" />

https://github.com/user-attachments/assets/52aed658-dfb7-4752-a250-473cfe03c0f3

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7008105
**Issue reported by:** NA
**Slack conversation:** NA
